source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx

```diff
@@ -0,0 +1,139 @@
+import { getEmptyLease } from '@/models/defaultInitializers';
+import { toTypeCode } from '@/utils/formUtils';
+import { render, RenderOptions } from '@/utils/test-utils';
+
+import { DetailFeeDetermination, IDetailFeeDeterminationProps } from './DetailFeeDetermination';
+
+const history = createMemoryHistory();
+
+describe('DetailFeeDetermination component', () => {
+  const setup = (
+    renderOptions: RenderOptions &
+      IDetailFeeDeterminationProps & { lease?: ApiGen_Concepts_Lease } = {},
+  ) => {
+    // render component under test
+    const component = render(
+      <Formik onSubmit={noop} initialValues={renderOptions.lease ?? getEmptyLease()}>
+        <DetailFeeDetermination
+          disabled={renderOptions.disabled}
+          nameSpace={renderOptions.nameSpace}
+        />
+      </Formik>,
+      {
+        ...renderOptions,
+        history,
+      },
+    );
+
+    return {
+      component,
+    };
+  };
+  it('renders minimally as expected', () => {
+    const { component } = setup({
+      lease: {
+        ...getEmptyLease(),
+        fileProperties: [
+          {
+            ...getEmptyPropertyLease(),
+            property: {
+              ...mockApiProperty,
+              areaUnit: toTypeCode('test'),
+              landArea: 123,
+            },
+            fileId: 1,
+          },
+        ],
+      },
+    });
+    expect(component.asFragment()).toMatchSnapshot();
+  });
+
+  it('renders a complete lease as expected', () => {
+    const { component } = setup({
+      lease: {
+        ...getEmptyLease(),
+        fileProperties: [
+          {
+            ...getEmptyPropertyLease(),
+            property: {
+              ...mockApiProperty,
+              areaUnit: toTypeCode('test'),
+              landArea: 123,
+            },
+          },
+        ],
+        amount: 1,
+        description: 'a test description',
+        programName: 'A program',
+        lFileNo: '222',
+        tfaFileNumber: '333',
+        psFileNo: '444',
+        motiName: 'test moti name',
+        note: 'a test note',
+        expiryDate: '2022-01-01',
+        isPublicBenefit: true,
+        isFinancialGain: false,
+        feeDeterminationNote: 'fee determination test note',
+        startDate: '2020-01-01',
+      },
+    });
+    expect(component.asFragment()).toMatchSnapshot();
+  });
+
+  it('renders the Public Benefit exists field', () => {
+    const {
+      component: { getByDisplayValue },
+    } = setup({
+      lease: {
+        ...getEmptyLease(),
+        isPublicBenefit: true,
+      },
+    });
+    expect(getByDisplayValue('Yes')).toBeVisible();
+  });
+
+  it('renders Financial Gain exists field', () => {
+    const {
+      component: { getByDisplayValue },
+    } = setup({
+      lease: {
+        ...getEmptyLease(),
+        isFinancialGain: true,
+      },
+    });
+    expect(getByDisplayValue('Yes')).toBeVisible();
+  });
+
+  it('renders the suggested Fee field', () => {
```

Comment:
since there is a calculation, I'd actually rather see more tests for this then the other fields. In general we don't need to write one test for field as that doesn't scale, but we want to write tests for non-trivial logic or business rules.

Reply:
https://confluence.th.gov.bc.ca/display/PIMS/Best+Practices#BestPractices-UnitTests

Reply:
I deleted the fields tests and increase the number of unit tests regarding the calculation of the suggested fee.